WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/BoldableTextView.swift

```diff
@@ -0,0 +1,113 @@
+import SwiftUI
+
+/// A text view that takes in a string whose substrings within two asterisks are bolded.
+///
+/// For example, initializing with a string "*Never* will I *stop learning*." will bold the substrings "Never" and "stop learning".
+///
+struct BoldableTextView: View {
+    fileprivate struct Element {
+        let content: String
+        let isBold: Bool
+
+        init(content: String, isBold: Bool) {
+            if isBold {
+                self.content = content.replacingOccurrences(of: boldRegionMark, with: "")
```

Comment:
One more improvement can be to pass text already without `boldRegionMark`. Our regular expression already has a matching group just for content `(.*?)`. We can leverage on that.

`NSTextCheckingResult` has an api to get a range at a certain index. Each of our matching results should have 2 ranges available: one for the whole match (that we use now) and one just for the content without the region mark. If we use that range to get the actual content, we don't need to replace occurrences later in the code.

Reply:
Great point! TIL'ed the 2 ranges in a `NSTextCheckingResult` - updated in https://github.com/woocommerce/woocommerce-ios/pull/3285/commits/78d1ea47625e0becf31e7242d76036c35516120d 😃